leader-election-impl/src/test/java/com/palantir/leader/proxy/AwaitingLeadershipProxyTest.java

```diff
@@ -264,6 +267,53 @@ public void shouldClearDelegateUponLosingLeadership() throws Exception {
         verify(mock).close();
     }
 
+    @Test
+    public void whenMaybeNotCurrentLeaderAndLeadershipNotLostShouldNotClearDelegate() throws Exception {
+        MyCloseable mockA = mock(MyCloseable.class);
+        LeadershipCoordinator leadershipCoordinator = LeadershipCoordinator.create(leaderElectionService);
+        MyCloseable proxyA =
+                AwaitingLeadershipProxy.newProxyInstance(MyCloseable.class, () -> mockA, leadershipCoordinator);
+
+        // Wait to gain leadership
+        Uninterruptibles.sleepUninterruptibly(Duration.ofMillis(100L));
+
+        doThrow(new MaybeNotCurrentLeaderException(TEST_MESSAGE)).when(mockA).val();
+
+        assertThatThrownBy(proxyA::val)
+                .isInstanceOf(MaybeNotCurrentLeaderException.class)
+                .hasMessage(TEST_MESSAGE);
+        verify(mockA, times(0)).close();
+    }
+
+    @Test
+    public void whenMaybeNotCurrentLeaderAndLeadershipLostShouldClearDelegate() throws Exception {
+        MyCloseable mockA = mock(MyCloseable.class);
+        LeadershipCoordinator leadershipCoordinator = LeadershipCoordinator.create(leaderElectionService);
+        MyCloseable proxyA =
+                AwaitingLeadershipProxy.newProxyInstance(MyCloseable.class, () -> mockA, leadershipCoordinator);
+
+        MyCloseable proxyB = AwaitingLeadershipProxy.newProxyInstance(
+                MyCloseable.class, () -> mock(MyCloseable.class), leadershipCoordinator);
+
+        // Wait to gain leadership
+        Uninterruptibles.sleepUninterruptibly(Duration.ofMillis(100L));
+
+        proxyA.val();
+        proxyB.val();
+
+        doThrow(new MaybeNotCurrentLeaderException("foo")).when(mockA).val();
+
+        loseLeadership(() -> {
+            proxyB.val();
+            return null;
```

Comment:
not actionable: This is pretty janky, though it's the test setup we already have, and I guess it works fine.